### ISSUE TITLE
Logging Architecture

### ISSUE THREAD

**Issue by lfrancke** [stackabletech/issues#202]
## Goal

This is done when we have a common CRD snippet to configure logging across the tools we support.

It should support the following features:
- It should be agnostic to the actual logging implementation (e.g. logback, log4j, Python Logging etc.) which means it _at least_ must support setting an overall log level as I believe this to be the common denominator. We also might want to support log level per "logger" (I believe all of them support a hierarchy).

- Further ideas welcome (e.g. size limit, format/pattern)

- It should allow configuration of log aggregation (via [Vector](https://vector.dev/)) including "reusable" sinks (e.g. configure my central Elasticsearch instance only once) and transport security

- It should allow for custom overrides of the underlying log config (e.g. log4j.properties) as well as the vector config (all guarantees go out of the window if our customers use it)

This should be worked on by at least two developers (preferably a python and a java logging "expert"), in order to identify the common denominator. 

The created CRD snippet should be extendible without breaking changes to the CRDs in the future (e.g. use complex Enums and/or generic datatypes/traits)

This needs to be presented at the architecture meeting. This can be done either via a document or example code in the operator-rs commons section.

Questions:
- Which parts should be user-pluggable/interchangable. Just the sink? The aggregator as well? How to integrate with existing infrastructure?
- Do we want to use the log agent as a sidecar or daemon?
- What does it look like for the user to configure logging?

## Roadmap

https://github.com/stackabletech/issues/issues/202#tasklist-block-ae3df65f-945a-49fc-9429-a5ca86d65260

Out of scope for now:
- Different configurations for `kubectl logs` and what ends up in kibana
- Logging kubernetes events

Links:
- Previous mini spike of Vector in ZooKeeper: https://github.com/stackabletech/zookeeper-operator/pull/450

**Comment by lfrancke** [comment]
@siegfriedweber This can be closed, I guess?

**Comment by siegfriedweber** [comment]
> @siegfriedweber This can be closed, I guess?

Yes, this ticket is completed.
